api/api.go

```diff
@@ -457,3 +475,296 @@ func fmtPublicKey(cert *x509.Certificate) string {
 	}
 	return fmt.Sprintf("%s %s", cert.PublicKeyAlgorithm, params)
 }
+
+type MockAuthority struct {
```

Comment:
I think you're mocking `LinkedAuthority` here, right? I would say rename it to something else (as discussed in a note above) and then is there an existing MockAuthority type that you can pull in. Kinda sucks to have to mock out all the Authority methods you're not planning on testing. Might be better to have a separate MockAuthority that you pull into this struct. Or a real struct that implements the Authority interface, just so you don't have to mock out all these unnecessary methods.

Reply:
Looks like we already had one of those in api_test.go. So, yea I would just split the mocks up into two. Is there any reason for them to be defined in the public facing files? Do you have tests that access this mock from other packages?

Reply:
Move the old one back to `api_test.go` and created a new `mockAdminAuthority` for the `Admin` APIs.